server/src/main/java/org/opensearch/cluster/coordination/JoinHelper.java

```diff
@@ -342,6 +348,7 @@ public void handleResponse(Empty response) {
                         pendingOutgoingJoins.remove(dedupKey);
                         logger.debug("successfully joined {} with {}", destination, joinRequest);
                         lastFailedJoinAttempt.set(null);
+                        nodeCommissioned.accept(true);
```

Comment:
In case the response from the leader in out of order for JoinRequest or waiting at the transport, the bool might not have right value. But I guess it will eventually get the right value in subsequent join attempts.

Reply:
That is correct